src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,562 @@
+            {
+                return 23;
+            }
+            else if (sampleMode == Sampling.UltraHighResolution)
+            {
+                return 44;
+            }
+            return 0;
+        }
+
+        /// <summary>
+        ///  Reads the pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Atmospheric pressure in Pa
+        /// </returns>
+        public async Task<double> ReadPressureAsync()
+        {
+            //Make sure the I2C device is initialized
+            if (!_initialized)
+            {
+                Begin();
+            }
+
+            if (ReadPowerMode() == PowerMode.Forced)
+            {
+                await Task.Delay(GetMeasurementTimeForForcedMode(ReadPressureSampling()));
+            }
+
+            //Read the temperature first to load the t_fine value for compensation
+            if (_temperatureFine == int.MinValue)
+            {
+                await ReadTemperatureAsync();
+            }
+
+            //Read the MSB, LSB and bits 7:4 (XLSB) of the pressure from the BMP280 registers
+            byte tmsb = Read8((byte)Registers.REGISTER_PRESSUREDATA_MSB);
+            byte tlsb = Read8((byte)Registers.REGISTER_PRESSUREDATA_LSB);
+            byte txlsb = Read8((byte)Registers.REGISTER_PRESSUREDATA_XLSB); // bits 7:4
+
+            //Combine the values into a 32-bit integer
+            int t = (tmsb << 12) + (tlsb << 4) + (txlsb >> 4);
+
+            //Convert the raw value to the pressure in Pa
+            long pres = BMP280_compensate_P_Int64(t);
+
+            //Return the temperature as a float value
+            return (pres) / 256;
+        }
+
+        /// <summary>
+        ///  Calculates the altitude in meters from the specified sea-level pressure(in hPa).
+        /// </summary>
+        /// <param name="seaLevelPressure" > 
+        ///  Sea-level pressure in hPa
+        /// </param>
+        /// <returns>
+        ///  Height in meters from the sensor
+        /// </returns>
+        public async Task<double> ReadAltitudeAsync(double seaLevelPressure)
+        {
+            //Make sure the I2C device is initialized
+            if (!_initialized)
+            {
+                Begin();
+            }
+
+            //Read the pressure first
+            var pressure = await ReadPressureAsync();
+            //Convert the pressure to Hectopascals(hPa)
+            pressure /= 100;
+
+            //Calculate and return the altitude using the international barometric formula
+            return 44330.0 * (1.0 - Math.Pow((pressure / seaLevelPressure), 0.1903));
+        }
+
+        /// <summary>
+        ///  Returns the temperature in DegC. Resolution is 0.01 DegC. Output value of “5123” equals 51.23 DegC.
+        /// </summary>
+        /// <param name="adcT"></param>
+        /// <returns>
+        ///  Degrees celsius
+        /// </returns>
+        private double BMP280_compensate_T_double(int adcT)
```

Comment:
Even though these two compensate methods are not part of the public surface area, I would try to name them a bit more readable, using PascalCasing instead of _ and using full words instead of just T for temperature or p for preassure